parmed/openmm/parameters.py

```diff
@@ -122,18 +141,180 @@ def write(self, dest, provenance=None):
             own_handle = False
 
         try:
-            if provenance is not None:
-                self._write_omm_provenance(provenance)
+            dest.write('<ForceField>\n')
+            self._write_omm_provenance(dest, original_file, reference)
+            self._write_omm_atom_types(dest)
             self._write_omm_residues(dest)
             self._write_omm_bonds(dest)
             self._write_omm_angles(dest)
-            self._write_omm_ubs(dest)
             self._write_omm_dihedrals(dest)
-            self._write_omm_periodic_impropers(dest)
             self._write_omm_impropers(dest)
-            self._write_omm_rb_torsions(dest)
+#           self._write_omm_rb_torsions(dest)
             self._write_omm_cmaps(dest)
-            self._write_omm_nonbonded(dest)
+#           self._write_omm_scripts(dest)
+#           self._write_omm_nonbonded(dest)
         finally:
+            dest.write('</ForceField>\n')
             if own_handle:
                 dest.close()
+
+    def _write_omm_provenance(self, dest, original_file, reference):
+        dest.write(' <Info>\n')
+        dest.write('  <Source>%s</Source>\n' % original_file)
```

Comment:
We should write the `md5` hash of the original file as well, since the source filename is insufficient to uniquely identify the file version.